add ca-certificates: for accessing remote raster files;

ImreSamu opened the pull request:
fix: https://github.com/postgis/docker-postgis/issues/307

important changes:
* `Dockerfile.template`
* `Dockerfile.alpine.template`
* `Dockerfile.master.template`

based on my local test;  minimal disk space increase is expected ( < 1MB )

```
# debian
new_postgis                                       14-3.2               77ce6bebee3d   20 seconds ago   575MB
postgis/postgis                                   14-3.2               9791b51b1ff7   4 days ago       575MB

# alpine
new_postgis                                       14-3.2-alpine        c18a9d1d4d26   3 minutes ago    395MB
postgis/postgis                                   14-3.2-alpine        128949707ecd   4 days ago       395MB

# master  - not tested .. disk space is less important.
```

disclaimer:  remote raster file access not tested;

IMHO: if the CI/CD is OK then ready to review/merge.

phillipross commented on 13-master/Dockerfile:
```diff
@@ -294,13 +294,17 @@ RUN set -ex \
+    #
+    # We keep ca-certificates: for accessing remote raster files
+    #    https://github.com/postgis/docker-postgis/issues/307
+    #
```
Is this correct?  The ca-certificates package should be removed in this case?

phillipross commented on 14-master/Dockerfile:
```diff
@@ -294,13 +294,17 @@ RUN set -ex \
+    #
+    # We keep ca-certificates: for accessing remote raster files
+    #    https://github.com/postgis/docker-postgis/issues/307
+    #
```
Is this correct?  The ca-certificates package should be removed in this case?

phillipross commented on 15beta3-master/Dockerfile:
```diff
@@ -294,13 +294,17 @@ RUN set -ex \
+    #
+    # We keep ca-certificates: for accessing remote raster files
+    #    https://github.com/postgis/docker-postgis/issues/307
+    #
```
Is this correct?  The ca-certificates package should be removed in this case?

phillipross commented on Dockerfile.master.template:
```diff
@@ -294,13 +294,17 @@ RUN set -ex \
+    #
+    # We keep ca-certificates: for accessing remote raster files
+    #    https://github.com/postgis/docker-postgis/issues/307
+    #
```
Is this correct?  The ca-certificates package should be removed in this case?

phillipross reviewed (commented):
should ca-certificats package be removed from master template?  or maybe i'm just looking at the changes incorrect?

phillipross commented on 13-master/Dockerfile:
```diff
@@ -327,5 +326,11 @@ RUN set -ex \
 COPY ./update-postgis.sh /usr/local/bin
-RUN cat /_pgis_full_version.txt
 
+RUN set -ex \
```
This only queries to see if the ca-certificates package is installed and reports the result, correct?  If the package is not installed, there's no place I see where the package gets installed.

phillipross commented on 14-master/Dockerfile:
```diff
@@ -327,5 +326,11 @@ RUN set -ex \
 COPY ./update-postgis.sh /usr/local/bin
-RUN cat /_pgis_full_version.txt
 
+RUN set -ex \
```
This only queries to see if the ca-certificates package is installed and reports the result, correct?  If the package is not installed, there's no place I see where the package gets installed.

phillipross commented on 15beta3-master/Dockerfile:
```diff
@@ -327,5 +326,11 @@ RUN set -ex \
 COPY ./update-postgis.sh /usr/local/bin
-RUN cat /_pgis_full_version.txt
 
+RUN set -ex \
```
This only queries to see if the ca-certificates package is installed and reports the result, correct?  If the package is not installed, there's no place I see where the package gets installed.

phillipross commented on Dockerfile.master.template:
```diff
@@ -327,5 +326,11 @@ RUN set -ex \
 COPY ./update-postgis.sh /usr/local/bin
-RUN cat /_pgis_full_version.txt
 
+RUN set -ex \
```
This only queries to see if the ca-certificates package is installed and reports the result, correct?  If the package is not installed, there's no place I see where the package gets installed.

phillipross reviewed (commented):
maybe i'm missing it, but in the master tenmplates it seems in cases where the ca-certificates package is not found to be installed, there's no explicit installation.  I'm not understanding the reason for the removal of the explicit ca-certificates package from the master templates.

ImreSamu commented:
> maybe i'm missing it, but in the master tenmplates it seems in cases where the ca-certificates package is not found to be installed, 
> there's no explicit installation. I'm not understanding the reason for the removal 
> of the explicit ca-certificates package from the master templates.


this is the "original" Dockerifle.master.templete ... ( search for 'ca-certificates' ) 
* https://github.com/postgis/docker-postgis/blob/a6a9f021e243c0b6f908cf8ad8d0ae5460dcb1b1/Dockerfile.master.template

this is the proposed "new" Dockerfile.master.template:
* https://github.com/postgis/docker-postgis/blob/468b99d9c101eecd00ac2058230ee0f6d68a1330/Dockerfile.master.template

And the 'ca-certificates' is installed in both of the "stage"  ( we are using "multi-stage builds" )
* stage1:  installed in L40  ( line 40 )
* stage2:  installed in L243  
* stage2:  uninstalled in L303 
      * I have removed this line from the new temple ... so we are keeping it.

in my proposal:  
I have 
* removed the "stage2" uninstall ( L303 in the original template ) 
* and added an "extra comment warning" + "extra package test" 
   *  .. for avoiding any non-perfect future refactoring.

phillipross commented:
oops, I see now.  I was definitely reading incorrectly 😅 looks good and thanks for the clarifications!
